include/EngineClient.h

```diff
@@ -0,0 +1,57 @@
+/*
+ * Engine.h - engine-system of LMMS
+ *
+ * Copyright (c) 2006-2014 Tobias Doerffel <tobydox/at/users.sourceforge.net>
+ *
+ * This file is part of LMMS - http://lmms.io
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program (see COPYING); if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301 USA.
+ *
+ */
+
+
+#ifndef ENGINECLIENT_H
+#define ENGINECLIENT_H
+
+#include "export.h"
+
+#include "Engine.h"
+
+class Song;
+class Mixer;
+class FxMixer;
+class BBTrackContainer;
+class ProjectJournal;
+
+class EXPORT EngineClient
+{
+public:
+	EngineClient( Engine * engine) : m_engine(engine) {}
+
+	LmmsCore * getEngine() const { return m_engine; }
```

Comment:
Is there a reason for using both `LmmsCore` and `Engine`? I think it makes sense to only use one or the other (probably `Engine`, as they are typedefs to the same type anyway.